src/V3Inst.cpp

```diff
@@ -501,15 +500,19 @@ class InstDeVisitor final : public VNVisitor {
                 return;
             }
         } else {
-            if (const AstUnpackArrayDType* const lhsarrp
-                = VN_CAST(nodep->lhsp()->dtypep()->skipRefp(), UnpackArrayDType)) {
-                if (const AstUnpackArrayDType* const rhsarrp
-                    = VN_CAST(nodep->rhsp()->dtypep()->skipRefp(), UnpackArrayDType)) {
+            if (const AstUnpackArrayDType* const rhsarrp
+                = VN_CAST(nodep->rhsp()->dtypep()->skipRefp(), UnpackArrayDType)) {
+                if (const AstUnpackArrayDType* const lhsarrp
+                    = VN_CAST(nodep->lhsp()->dtypep()->skipRefp(), UnpackArrayDType)) {
                     // copy between arrays
                     if (!VN_IS(lhsarrp->subDTypep()->skipRefp(), IfaceRefDType)) return;
                     if (!VN_IS(rhsarrp->subDTypep()->skipRefp(), IfaceRefDType)) return;
                     if (VN_AS(rhsarrp->subDTypep()->skipRefp(), IfaceRefDType)->isVirtual())
                         return;
+                    if (!VN_AS(lhsarrp->subDTypep()->skipRefp(), IfaceRefDType)->isVirtual()) {
+                        nodep->lhsp()->v3error("Illegal target of assignment");
```

Comment:
Illegal or unsupported? Can it turn up if something unexpected like a queue is involved?  Add the types to the error so more obvious?   (prettyDTypeNameQ())

Reply:
Yes, it seems like it would be better described as unsupported.